specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+
+// The vote accounting for a proposal.
+mapping(uint256 proposalId => ProposalVote votes) internal _proposalVotes;
+
+// The proposals that have been created.
+mapping(uint256 proposalId => ProposalCore proposal) internal _proposals;
+```
+
+## Types
+
+The `Governor` contract MUST define the following types:
+
+### `ProposalVote`
+
+`ProposalVote` defines the vote accounting for a proposal, encapsulated in the following struct:
+
+```solidity
+struct ProposalVote {
+  uint256 againstVotes;
+  uint256 forVotes;
+  uint256 abstainVotes;
+  mapping(address => bool) hasVoted;
+}
+```
+
+| Name                     | Type            | Description                                                             |
+|--------------------------|-----------------|-------------------------------------------------------------------------|
+| `againstVotes`          | `uint256` | Amount of votes against the proposal.                         |
+| `forVotes`              | `uint256`       | Amount of votes in favor of the proposal.                |
+| `abstainVotes`                 | `uint256`       | Amount of votes abstaining.   |
+| `hasVoted`                 | `mapping`       | Whether an account has voted for this proposal or not.   |
+
+### `ProposalCore`
+
+`ProposalCore` defines the core information of a proposal, encapsulated in the following struct:
+
+```solidity
+struct ProposalCore {
+  uint64 voteStart;
+  uint64 voteEnd;
+  bool executed;
+  bool canceled;
+  address votingModule;
+  uint8 proposalType;
+}
+```
+
+| Name                     | Type            | Description                                                             |
+|--------------------------|-----------------|-------------------------------------------------------------------------|
+| `voteStart`          | `uint64` | The block timestamp when voting started.                         |
+| `voteEnd`              | `uint64`       | The block timestamp when voting ends.                |
+| `executed`                 | `bool`       | Indicator for proposal execution.   |
+| `canceled`                 | `bool`       | Indicator for proposal cancellation.   |
+| `votingModule`                 | `address`       | The voting module of the proposal. Zero address if none.   |
+| `proposalType`                 | `uint8`       | The type of proposal.   |
+
+## Proposal Types
+
+The `Governor` contract uses proposal types to define the quorum and approval threshold for a proposal to pass.
+These proposal types are created and stored as part of an external contract called `ProposalTypesConfigurator`.
+The `Governor` contract can only create or edit proposal types, and the amount of proposal types is limited to 255.
+
+With the [`proposeWithModule`](#proposewithmodule) function, proposers can create proposals with any supported proposal
+type.
+
+## Modules
+
+Additionally, from proposal types, the `Governor` contract allows to use modules as part of the proposal logic.
```

Comment:
What is the interface for a module?

Reply:
added in https://github.com/ethereum-optimism/specs/pull/315/commits/1c47afea51f6c7b5c13d777e0a2f6209cb7ab034